src/testnet/helium/run_loop.rs

```diff
@@ -0,0 +1,262 @@
+use super::{Config, Node, BurnchainSimulator, BurnchainState, LeaderTenure};
+
+use chainstate::burn::{ConsensusHash};
+use chainstate::stacks::db::{StacksHeaderInfo, StacksChainState, ClarityTx};
+use chainstate::burn::{BlockHeaderHash};
+use chainstate::stacks::{StacksBlock, TransactionAuth, TransactionSpendingCondition, SinglesigSpendingCondition, TransactionPayload};
+use chainstate::stacks::events::StacksTransactionReceipt;
+use burnchains::Address;
+
+use util::sleep_ms;
+
+/// RunLoop is coordinating a simulated burnchain and some simulated nodes
+/// taking turns in producing blocks.
+pub struct RunLoop {
+    config: Config,
+    node: Node,
+    new_burnchain_state_callback: Option<fn(u64, &BurnchainState)>,
+    new_tenure_callback: Option<fn(u64, &LeaderTenure)>,
+    new_chain_state_callback: Option<fn(u64, &mut StacksChainState, StacksBlock, StacksHeaderInfo, Vec<StacksTransactionReceipt>)>,
+}
+
+#[allow(unused_macros)]
+macro_rules! info_blue {
+    ($($arg:tt)*) => ({
+        eprintln!("\x1b[0;96m{}\x1b[0m", format!($($arg)*));
```

Comment:
Please open an issue to do the following:

* Change the escape codes to constants
* Use `isatty(2)` to only print the escape codes if `stderr` is a TTY (otherwise, if you're logging this to a file, the escape codes will be written, too.

Reply:
Issue available here: https://github.com/blockstack/stacks-blockchain/issues/1391